File: leetcode/medium/1551. Minimum Operations to Make Array Equal.ts
function minOperations(n: number): number {
  if (n === 1) return 0;
  const arr: number[] = new Array(n);
  for (let i = 0; i < n; i += 1) {
    arr[i] = 2 * i + 1;
  }
  // making the first and the last one the same number first
  const mid_num: number = (arr[0] + arr[n - 1]) / 2;
  // all the number will be the same as mid_num
  let count_operation: number = 0;
  for (let i = 0; i < arr.length / 2; i += 1) {
    count_operation += mid_num - arr[i];
  }
  return count_operation;
}

File: leetcode/medium/1551. Minimum Operations to Make Array Equal.rs
impl Solution {
    pub fn min_operations(n: i32) -> i32 {
        let mut count_operation: i32 = 0;
        if n == 1 {
            return count_operation;
        }
        let mid_num: i32 = (1 + ((n-1)*2+1))/2;
        for i in 0..n/2 as i32{
            count_operation += (mid_num - (i*2+1));
        }
        return count_operation;
    }
}